### ISSUE TITLE
[BUG] cannot downgrade from 87863

### ISSUE THREAD

**Issue by exa-byte** [DJSchaffner/aoe2de_patcher#7]
**Describe the bug**
I just tried to revert back to the last patch 85208 and get folling error message. Since the patch is relatively new I thought I notify you. If you already know about it just close the issue.

**To Reproduce**
1. Download aoe2de_patcher
2. Run the patcher
3. Select game directory and target version 85208
4. Receive console output:
```
Game directory set to: E:\Program Files (x86)\Steam\SteamApps\common\AoE2DE
Installed version detected: 87863
Starting download phase
Generating list of changes
The installed version currently doesn't support downgrading. Please be patient or notify me on GitHub!
Error during download!
Could not patch!
```

**Comment by DJSchaffner** [comment]
Hey there, thank you for notifying me :)
Looks like I made a mistake in the patch before this one which broke the patching. Should be fixed in a few minutes. Please report back if it's still not working for you

**Comment by exa-byte** [comment]
Hi, I think the original bug is fixed and its working now. 

However I have another problem with steam guard, as i got promted too many times (after each manifest?) and now the login limit is reached. 
```
Game directory set to: E:\Program Files (x86)\Steam\SteamApps\common\AoE2DE
Installed version detected: 87863
Starting download phase
Generating list of changes
Connecting to Steam3... Done!
Logging 'my_steam_user_name' into Steam3...Disconnected from Steam
This account is protected by Steam Guard.
Please enter the authentication code sent to your email address: Retrying Steam3 connection... Done!
Logging 'my_steam_user_name' into Steam3... Done!
Using Steam3 suggested CellID: 90
Got 12 licenses for account!
Got AppInfo for 813780
Using app branch: 'Public'.
Got depot key for 813781 result: OK
Processing depot 813781 - 
Downloading depot manifest...Got manifest request code for 813781 1866078048582611873 result: 12927770432423408344
 Done!
Manifest 1866078048582611873 (6/23/2023 10:05:55 AM)
Total downloaded: 0 bytes (0 bytes uncompressed) from 1 depots
Disconnected from Steam
Connecting to Steam3... Done!
Logging 'my_steam_user_name' into Steam3...Disconnected from Steam
This account is protected by Steam Guard.
Please enter the authentication code sent to your email address: Retrying Steam3 connection... Done!
Logging 'my_steam_user_name' into Steam3... Done!
Using Steam3 suggested CellID: 90
Got 12 licenses for account!
Got AppInfo for 813780
Using app branch: 'Public'.
Got depot key for 813781 result: OK
Processing depot 813781 - 
Downloading depot manifest...Got manifest request code for 813781 7121668865398588908 result: 435102769771737615
 Done!
Manifest 7121668865398588908 (6/2/2023 2:11:45 AM)
Total downloaded: 0 bytes (0 bytes uncompressed) from 1 depots
Disconnected from Steam
Connecting to Steam3... Done!
Logging 'my_steam_user_name' into Steam3...Disconnected from Steam
This account is protected by Steam Guard.
Please enter the authentication code sent to your email address: Retrying Steam3 connection... Done!
Logging 'my_steam_user_name' into Steam3... Done!
Using Steam3 suggested CellID: 90
Got 12 licenses for account!
Got AppInfo for 813780
Using app branch: 'Public'.
Got depot key for 813782 result: OK
Processing depot 813782 - 
Downloading depot manifest...Got manifest request code for 813782 2748266964518663227 result: 10747849037557407625
 Done!
Manifest 2748266964518663227 (6/20/2023 5:08:07 AM)
Total downloaded: 0 bytes (0 bytes uncompressed) from 1 depots
Disconnected from Steam
Connecting to Steam3... Done!
Logging 'my_steam_user_name' into Steam3...Disconnected from Steam
This account is protected by Steam Guard.
Please enter the authentication code sent to your email address: Retrying Steam3 connection... Done!
Logging 'my_steam_user_name' into Steam3... Done!
Using Steam3 suggested CellID: 90
Got 12 licenses for account!
Got AppInfo for 813780
Using app branch: 'Public'.
Got depot key for 813782 result: OK
Processing depot 813782 - 
Downloading depot manifest...Got manifest request code for 813782 3357497345259998783 result: 11233523082943270109
 Done!
Manifest 3357497345259998783 (5/29/2023 9:13:33 AM)
Total downloaded: 0 bytes (0 bytes uncompressed) from 1 depots
Disconnected from Steam
Connecting to Steam3... Done!
Logging 'my_steam_user_name' into Steam3...Disconnected from Steam
This account is protected by Steam Guard.
Please enter the authentication code sent to your email address: Retrying Steam3 connection... Done!
Logging 'my_steam_user_name' into Steam3... Done!
Using Steam3 suggested CellID: 90
Got 12 licenses for account!
Got AppInfo for 813780
Using app branch: 'Public'.
Got depot key for 813783 result: OK
Processing depot 813783 - 
Downloading depot manifest...Got manifest request code for 813783 1541580786568448903 result: 13586762337088069563
 Done!
Manifest 1541580786568448903 (6/2/2023 5:05:37 AM)
Total downloaded: 0 bytes (0 bytes uncompressed) from 1 depots
Disconnected from Steam
Connecting to Steam3... Done!
Logging 'my_steam_user_name' into Steam3...Disconnected from Steam
This account is protected by Steam Guard.
Please enter the authentication code sent to your email address: Retrying Steam3 connection... Done!
Logging 'my_steam_user_name' into Steam3... Done!
Using Steam3 suggested CellID: 90
Got 12 licenses for account!
Got AppInfo for 813780
Using app branch: 'Public'.
Got depot key for 813783 result: OK
Processing depot 813783 - 
Downloading depot manifest...Got manifest request code for 813783 4263429827111208216 result: 6847110655795213675
 Done!
Manifest 4263429827111208216 (5/24/2023 9:02:24 AM)
Total downloaded: 0 bytes (0 bytes uncompressed) from 1 depots
Disconnected from Steam
Connecting to Steam3... Done!
Logging 'my_steam_user_name' into Steam3...Disconnected from Steam
This account is protected by Steam Guard.
Please enter the authentication code sent to your email address: Retrying Steam3 connection... Done!
Logging 'my_steam_user_name' into Steam3... Done!
Using Steam3 suggested CellID: 90
Got 12 licenses for account!
Got AppInfo for 813780
Using app branch: 'Public'.
Got depot key for 813784 result: OK
Processing depot 813784 - 
Downloading depot manifest...Got manifest request code for 813784 1077533377650376618 result: 13847020094661505117
 Done!
Manifest 1077533377650376618 (6/14/2023 2:28:46 PM)
Total downloaded: 0 bytes (0 bytes uncompressed) from 1 depots
Disconnected from Steam
Connecting to Steam3... Done!
Logging 'my_steam_user_name' into Steam3...Disconnected from Steam
This account is protected by Steam Guard.
Please enter the authentication code sent to your email address: Retrying Steam3 connection... Done!
Logging 'my_steam_user_name' into Steam3... Done!
Using Steam3 suggested CellID: 90
Got 12 licenses for account!
Got AppInfo for 813780
Using app branch: 'Public'.
Got depot key for 813784 result: OK
Processing depot 813784 - 
Downloading depot manifest...Got manifest request code for 813784 7882198738898788470 result: 6159701792748240796
 Done!
Manifest 7882198738898788470 (4/20/2023 10:22:08 AM)
Total downloaded: 0 bytes (0 bytes uncompressed) from 1 depots
Disconnected from Steam
Connecting to Steam3... Done!
Logging 'my_steam_user_name' into Steam3...Disconnected from Steam
This account is protected by Steam Guard.
Please enter the authentication code sent to your email address: Retrying Steam3 connection... Done!
Logging 'my_steam_user_name' into Steam3... Done!
Using Steam3 suggested CellID: 90
Got 12 licenses for account!
Got AppInfo for 813780
Using app branch: 'Public'.
Got depot key for 813787 result: OK
Processing depot 813787 - 
Downloading depot manifest...Got manifest request code for 813787 7470002296534019464 result: 16355323818162720165
 Done!
Manifest 7470002296534019464 (6/6/2023 6:55:45 AM)
Total downloaded: 0 bytes (0 bytes uncompressed) from 1 depots
Disconnected from Steam
Connecting to Steam3... Done!
Logging 'my_steam_user_name' into Steam3...Disconnected from Steam
This account is protected by Steam Guard.
Please enter the authentication code sent to your email address: Retrying Steam3 connection... Done!
Logging 'my_steam_user_name' into Steam3...Unable to login to Steam3: RateLimitExceeded
Disconnected from Steam
Unable to get steam3 credentials.
Error: InitializeSteam failed
```

**Comment by DJSchaffner** [comment]
The login restriction ends after about 30 minutes, I think. However, this does look a bit odd. Also, there is shouldn't be that many requests happening, and usually it should ask for the code once and then keeps the session for the next depot, so it doesn't have to ask again.  

Steam might have changed stuff again with their 2FA, so it broke something. I will look into it.   

As a workaround, you can use another account and family share aoe to it. If you don't enable 2FA there you only have to enter a code from an email once.

**Comment by DJSchaffner** [comment]
Okay, after trying around a bit. I had to approve my login in the authenticator app on my phone. After that it kept the session and I didn't have to enter the 2FA code anymore.  
See if that is maybe the case for you as well

**Comment by exa-byte** [comment]
> Okay, after trying around a bit. I had to approve my login in the authenticator app on my phone. After that it kept the session and I didn't have to enter the 2FA code anymore. See if that is maybe the case for you as well

Hi,
can you describe how you approved the login? For me there is only a push notification with the code that works for just one request. It doesn't get saved though.

**Comment by DJSchaffner** [comment]
I tried to trigger it earlier, but couldn't find a way to do it consistently. It just showed up as an option on my phone at some point :/
Also, when I removed the granted access to the app from my phone to test it a bit more, it seems like it's now blocked or something, so I can't test anymore until it's unblocked again with that account.   

I made some changes to the tool though and will make a release in the next few minutes. Might help with the login, I don't think it will change much, though. But I added a few popups to know when it's done or stopped working though, which might be useful.
